src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt

```diff
@@ -0,0 +1,676 @@
+
+    override val isFabricVisible by lazy {
+        selectedMinecraftVersionFlow.map { version ->
+            !fabricDisabledMCVersions.contains(version)
+        }
+    }
+
+    override val isLegacyFabricVisible: Flow<Boolean> by lazy {
+        selectedMinecraftVersionFlow.map { version ->
+            !legacyFabricDisabledMCVersions.contains(version)
+        }
+    }
+
+    override val isForgeVisible by lazy {
+        selectedMinecraftVersionFlow.map { version ->
+            !forgeDisabledMCVersions.contains(version)
+        }
+    }
+
+    override val isQuiltVisible by lazy {
+        selectedMinecraftVersionFlow.map { version ->
+            !quiltDisabledMCVersions.contains(version)
+        }
+    }
+
+
+    override val loaderTypeFabricSelected: Flow<Boolean> = selectedLoaderType.map {
+        it == LoaderType.FABRIC
+    }
+    override val loaderTypeFabricEnabled = MutableStateFlow(true)
+
+    override val loaderTypeForgeSelected: Flow<Boolean> = selectedLoaderType.map {
+        it == LoaderType.FORGE
+    }
+    override val loaderTypeLegacyFabricSelected: Flow<Boolean> =
+        selectedLoaderType.map {
+            it == LoaderType.LEGACY_FABRIC
+        }
+    override val loaderTypeForgeEnabled = MutableStateFlow(true)
+    override val loaderTypeLegacyFabricEnabled = MutableStateFlow(true)
+
+    override val loaderTypeNoneSelected: Flow<Boolean> = selectedLoaderType.map {
+        it == null
+    }
+    override val loaderTypeNoneEnabled = MutableStateFlow(true)
+
+    override val loaderTypeQuiltSelected: Flow<Boolean> = selectedLoaderType.map {
+        it == LoaderType.QUILT
+    }
+    override val loaderTypeQuiltEnabled = MutableStateFlow(true)
+
+    /**
+     * Just a small copy function, to copy the values of a map.
+     *
+     * This is useful for flows, as a map is considered a persistent structure.
+     * Thus, modifications to the map doesn't "change" the map.
+     */
+    private fun <K, V> Map<K, V>.copy(): Map<K, V> {
+        val map = HashMap<K, V>()
+        this.forEach { (k, v) ->
+            map[k] = v
+        }
+        return map
+    }
+
+    private fun install(isServer: Boolean) {
+        val installable: Installable
+        try {
+            val selectedLoaderVersion = selectedLoaderVersion.value
+            val selectedMinecraftVersion = selectedMinecraftVersionFlow.value ?: return
+            val description = description.value
+            val name = name.value
+
+            installable = VanillaInstallable(
+                MinecraftManager.getMinecraftVersion(selectedMinecraftVersion),
+                selectedLoaderVersion,
+                description
+            )
+            installable.instanceName = name
+            installable.isReinstall = false
+            installable.isServer = isServer
+            val success = installable.startInstall()
+            if (success) {
+                // TODO Reset server
```

Comment:
Is this still a TODO?

Reply:
I believe this was what we discussed on discord about resetting the UI after an install is done. So no, I will remove the TODO.